### PR TITLE
fix: removing party invitations

### LINKED ISSUE #7793: party invitations can't be removed
Reported by Kingkrimmson (cea63e37-cf21-4296-a2d5-416dd8664e9a) in the Report a Bug guild:

**Hey, I'm a party leader and noticed that when I try to "Boot Member" for someone who has been invited but hasn't join yet (effectively to rescind invite), they are removed from the list, but when I come back to the party screen, they're still listed. Is this a known issue, or am I missing something?**

I've seen similar comments before but looks like we don't have an issue for it. I haven't tested yet to see if it's reproducable. If anyone seeing this would like to test and comment, please do, otherwise I will later when I have some free time.

### PR DESCRIPTION
**Pull request by Hus274**
closes #7793 
### Changes

Fixed mistake party invitation removal code. Also fixed bug in object generator that prevented the invitees of a group from being correctly populated. 

Also did some test driven development to figure out why the existing test didn't catch this behavior before. It looks like the test was expecting the party invitations to be an array instead of an object.

In looking in the invite groups method in groups of the server controllers, I noticed that the post request awaits the return value of _inviteToGroup but it wasn't returning anything before so I fixed that as well.

---

UUID: 2a4d30ee-4833-420e-be7c-ad6a6e261c5f

### CODE REVIEW

File: test/helpers/api-integration/v3/object-generators.js [hunk 1]
```diff
@@ -128,6 +128,10 @@ export async function createAndPopulateGroup (settings = {}) {
 
   await Bluebird.all(invitationPromises);
 
+  await Bluebird.all(invitees.map((invitee) => {
```
Comment (crookedneighbor) [inline test/helpers/api-integration/v3/object-generators.js]: You could do 

``` js
await Bluebird.map(invitees, (invitee) => invitee.sync())
```

File: website/server/controllers/api-v3/groups.js [hunk 2]
```diff
@@ -514,6 +514,7 @@ async function _inviteByUUID (uuid, group, inviter, req, res) {
     }
 
     userToInvite.invitations.party = {id: group._id, name: group.name, inviter: inviter._id};
+    return userToInvite.save();
```
Comment (crookedneighbor) [inline website/server/controllers/api-v3/groups.js]: Returning here skips all the email stuff that happens in the background.

Comment (Hus274) [inline website/server/controllers/api-v3/groups.js]: Meant to put it at the bottom of the function, that was my mistake. I'll fix this.

**Comment by Hus274** [comment]
Updated that according to the comments. Should be good to go now!

**Comment by audiolion** [comment]
tests and changes look good to me!

**Comment by crookedneighbor** [comment]
Notated toward your next tier.

**Pull request merged**